python/cudf/cudf/core/column_accessor.py

```diff
@@ -249,9 +249,7 @@ def to_pandas_index(self) -> pd.Index:
             # prevents coercion of values to a different type
             # (e.g., ''->NaT)
             result = pd.MultiIndex.from_frame(
-                pd.DataFrame(
-                    self.names, columns=self.level_names, dtype="object"
-                ),
```

Comment:
Is the `dtype="object"` needed to prevent the kind of coercion alluded to in the comment above?

Reply:
I don't think so. AFAICT making a pandas `DataFrame` (with typed columns) and using `from_frame` is what's needed here. Specifying `dtype="object"` is problematic because pandas plans to stop inferring dtypes when a MultiIndex is constructed from a `DataFrame` where all the dtypes are `object`. Do you have an example of a case that we should test and verify?

Reply:
I think the comment refers to cases like the one below:

```python
In [107]: ca = cudf.core.column_accessor.ColumnAccessor({'': [], pd.Timestamp('2001-01-01'): []})

In [108]: pd.MultiIndex.from_frame(pd.DataFrame(ca.names))  # '' becomes 'NaT'
<warning snipped>
Out[108]: 
MultiIndex([(       'NaT',),
            ('2001-01-01',)],
           names=[0])

In [110]: pd.MultiIndex.from_frame(pd.DataFrame(ca.names, dtype='object'))  # '' stays ''
Out[110]: 
MultiIndex([(                 '',),
            (2001-01-01 00:00:00,)],
           names=[0])
```

Reply:
Done. I also inserted an assertion that the pandas version is less than 2.0 to make sure that we revisit this when the time comes to upgrade.